File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-2.java
class Solution {
    public int characterReplacement(String s, int k) {
        int left = 0;
        int[] count = new int[26];
        int max = 0;
        int maxCount = 0;
        for (int right = 0; right < s.length(); right++) {
            count[s.charAt(right) - 'A']++;
            maxCount = Math.max(maxCount, count[s.charAt(right) - 'A']);
            while ((right - left + 1) - maxCount > k) {
                count[s.charAt(left) - 'A']--;
                left++;
                maxCount = Math.max(maxCount, count[s.charAt(left) - 'A']);
            }
            max = Math.max(right - left + 1, max);
        }
        return max;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-2.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int n = heights.size();
        int left = 0;
        int right = n - 1;
        int ans = 0;
        while (left < right) {
            int leftHeight = heights[left];
            int rightHeight = heights[right];
            int area = (right - left) * min(leftHeight, rightHeight);
            ans = max(area, ans);
            if (leftHeight < rightHeight) {
                left++;
            } else {
                right--;
            }
        }
        return ans;
    }
};
